src/cmd/tools/m3ctl/main/apply/cmd.go

```diff
@@ -0,0 +1,108 @@
+package apply
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/m3db/m3/src/cmd/tools/m3ctl/main/errors"
+	"github.com/m3db/m3/src/cmd/tools/m3ctl/main/globalopts"
+	"github.com/m3db/m3/src/x/config/configflag"
+)
+
+type applyVals struct {
+	yamlFlag *configflag.FlagStringSlice
+}
+type Context struct {
+	vals       *applyVals
+	handlers   applyHandlers
+	GlobalOpts globalopts.GlobalOpts
+	Flags      *flag.FlagSet
+}
+type applyHandlers struct {
+	handle func(*applyVals, globalopts.GlobalOpts) error
+}
+
+func InitializeFlags() Context {
+	return _setupFlags(
+		&applyVals{
+			yamlFlag: &configflag.FlagStringSlice{},
+		},
+		applyHandlers{
+			handle: doApply,
+		})
+}
+
+func _setupFlags(finalArgs *applyVals, handlers applyHandlers) Context {
```

Comment:
any particular reason not to use `github.com/spf13/cobra`? you're writing a lot of boilerplate that could be eliminated by using a higher-level package that's already included

Reply:
standard bad reason - it started small and was simple, and its already at the point where the next step is to wrap it up in interfaces and make it all fancy.  Yea, its time to go to cobra.

Reply:
i've switched to cobra